### PR TITLE
[1LP][RFR] WaitTab and wt.core update for Table

### PR DESCRIPTION
**Pull request by mshriver**
testing table on tab fix

{{ pytest: cfme/tests/cloud_infra_common/test_provisioning.py cfme/tests/candu/test_host_graph.py --use-provider default --long-running -vvv }}

part of fixing #7960

### CODE REVIEW

**Comment by mshriver** [comment]
@digitronik note Jenkins is running this branch as its default for 595 at the moment. PRT will take quite a while to get through even the subset of tests I have included, but its ran about 20 so far (as of writing). There are test failures, but they aren't necessarily related to this change. There will be followup PRs to deal with test specific failures, or other View related failures. Thanks for review!

File: cfme/configure/tasks.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ class Task(BaseEntity):
         name: Name of the task.
         user: User who initiated the task.
         server: Cfme server.
-        tab: Tab where current task located.
+        tab: WaitTab where current task located.
```
Comment (izapolsk) [inline cfme/configure/tasks.py]: unintentional rename ?

File: widgetastic_manageiq/__init__.py [hunk 2]
```diff
@@ -3660,7 +3660,31 @@ def fill(self, value):
         return True
 
 
-class PotentiallyInvisibleTab(Tab):
+class WaitTab(Tab):
```
Comment (izapolsk) [inline widgetastic_manageiq/__init__.py]: please add a docstring and some todo. we need to fix ensure page safe later

File: widgetastic_manageiq/__init__.py [hunk 3]
```diff
@@ -3660,7 +3660,31 @@ def fill(self, value):
...
+        _to_try = 2
+        _tries = 0
+        while _tries < _to_try:
+            _tries += 1
+            self.select()
+            try:
+                wait_for(self.is_active, delay=.1, num_sec=2)
+            except TimedOutError:
+                self.logger.exception(
+                    "Tab was not active in 3s{}".format(", retrying" if _tries < _to_try else "")
```
Comment (jawatts) [inline widgetastic_manageiq/__init__.py]: should this be 2s?

Comment (mshriver) [reply widgetastic_manageiq/__init__.py]: yeah

File: widgetastic_manageiq/__init__.py [hunk 4]
```diff
@@ -3660,7 +3660,31 @@ def fill(self, value):
...
+    def child_widget_accessed(self, widget):
+        _to_try = 2
+        _tries = 0
+        while _tries < _to_try:
+            _tries += 1
+            self.select()
+            try:
+                wait_for(self.is_active, delay=.1, num_sec=2)
+            except TimedOutError:
+                self.logger.exception(
```
Comment (izapolsk) [inline widgetastic_manageiq/__init__.py]: warn ?

File: widgetastic_manageiq/__init__.py [hunk 5]
```diff
@@ -3660,7 +3660,31 @@ def fill(self, value):
         return True
 
 
-class PotentiallyInvisibleTab(Tab):
+class WaitTab(Tab):
+    def child_widget_accessed(self, widget):
+        _to_try = 2
+        _tries = 0
+        while _tries < _to_try:
```
Comment (izapolsk) [inline widgetastic_manageiq/__init__.py]: 2 attempts ? you mentioned 3 when you talked about it.

File: widgetastic_manageiq/__init__.py [hunk 6]
```diff
@@ -3660,7 +3660,31 @@ def fill(self, value):
...
+            try:
+                wait_for(self.is_active, delay=.1, num_sec=2)
+            except TimedOutError:
+                self.logger.exception(
+                    "Tab was not active in 3s{}".format(", retrying" if _tries < _to_try else "")
+                )
+                if _tries >= _to_try:  # raise the TimedOutError if we've tried enough times
+                    raise
+            else:
+                break  # no retry
```
Comment (izapolsk) [inline widgetastic_manageiq/__init__.py]: you could put break right after wait

File: widgetastic_manageiq/__init__.py [hunk 7]
```diff
@@ -3660,7 +3660,31 @@ def fill(self, value):
...
+        try:
+            widget.wait_displayed(timeout=2)
+        except TimedOutError:
+            self.logger.exception("Tab widget not displayed, continuing...")
+            pass
```
Comment (izapolsk) [inline widgetastic_manageiq/__init__.py]: not necessary

File: widgetastic_manageiq/__init__.py (hunk 5 above)
Comment (mshriver) [reply widgetastic_manageiq/__init__.py]: Just two tries, don't want it taking up more time than that for workaround

File: widgetastic_manageiq/__init__.py (hunk 4 above)
Comment (mshriver) [reply widgetastic_manageiq/__init__.py]: could, exception log gives us the exception as well as the passed message.

File: widgetastic_manageiq/__init__.py (hunk 6 above)
Comment (mshriver) [reply widgetastic_manageiq/__init__.py]: wanted to keep 1 statement in the try when writing, but yep going to collapse this now. Thanks!

**Pull request merged**